src/components/receive/Receive.tsx

```diff
@@ -0,0 +1,220 @@
+  const [copied, setCopied] = useState(false)
+  const [amountDisplayMode, setAmountDisplayMode] = useState<'sats' | 'btc'>('sats')
+
+  const { jars } = useJamDisplayContext()
+  const client = useApiClient()
+  const session = getSession()
+
+  const totalBalance = useMemo(() => {
+    return jars.reduce((total, jar) => total + (jar.balance || 0), 0)
+  }, [jars])
+
+  const walletFileName = session?.walletFileName
+
+  const getAddressQuery = useQuery({
+    ...getaddressOptions({
+      client,
+      path: {
+        walletname: walletFileName!,
+        mixdepth: String(selectedJarIndex),
+      },
+    }),
+    retry: false,
+    enabled: walletFileName !== undefined && selectedJarIndex !== undefined,
+    staleTime: 1,
+  })
+
+  useEffect(() => {
+    if (getAddressQuery.data === undefined) return
+    setBitcoinAddress(getAddressQuery.data.address)
+  }, [getAddressQuery.data])
+
+  const isQrLoading = useMemo(() => {
+    return getAddressQuery.isFetching
+  }, [getAddressQuery.isFetching])
+
+  const copyToClipboard = () => {
+    if (bitcoinAddress) {
+      navigator.clipboard.writeText(bitcoinAddress)
+      setCopied(true)
+      toast.success('Bitcoin address copied to clipboard')
+    } else {
+      toast.error('No Bitcoin address to copy')
+    }
+  }
+
+  const shareAddress = () => {
+    if (navigator.share && bitcoinAddress) {
+      navigator
+        .share({
+          title: 'Bitcoin Address',
+          text: bitcoinAddress,
+        })
+        .catch(() => {
+          // Fall back to clipboard if sharing fails
+          copyToClipboard()
+        })
+    } else if (bitcoinAddress) {
+      copyToClipboard()
+    } else {
+      toast.error('No Bitcoin address to share')
+    }
+  }
+
+  const getNewAddress = useCallback(() => {
+    getAddressQuery.refetch()
+  }, [getAddressQuery])
+
+  const selectJar = (index: number) => {
+    setSelectedJarIndex(index)
+  }
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value
+    if (value === '') {
+      setAmount(undefined)
+      return
+    }
+
+    const numValue = parseFloat(value)
+    if (amountDisplayMode === 'btc') {
+      setAmount(btcToSats(numValue.toString()))
+    } else {
+      setAmount(Math.floor(numValue))
+    }
+  }
+
+  const toggleDisplayMode = () => {
+    setAmountDisplayMode((prev) => (prev === 'sats' ? 'btc' : 'sats'))
+  }
+
+  const getDisplayAmount = () => {
+    if (!amount) return ''
+    if (amountDisplayMode === 'btc') {
+      return satsToBtc(amount.toString()).toFixed(8)
+    }
+    return amount.toString()
+  }
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setCopied(false)
+    }, 5000)
+
+    return () => clearTimeout(timer)
+  }, [copied])
+
+  const selectedJar = jars[selectedJarIndex]
+  const hasValidJar = selectedJar && selectedJar.account !== undefined
+
+  if (!hasValidJar || !walletFileName) {
```

Comment:
Wallet File name should always be present here, right?
Maybe we should add it as an input to the component itself?

What do you think?